crates/hulk/src/webots/keyboard_device.rs

```diff
@@ -1,28 +1,38 @@
 use types::TouchSensors;
-use webots::{Keyboard, Robot};
+use webots::{Keyboard, Receiver, Robot};
 
 use super::interface::SIMULATION_TIME_STEP;
 
 pub struct KeyboardDevice {
     keyboard: Keyboard,
+    receiver: Receiver,
 }
 
 impl Default for KeyboardDevice {
     fn default() -> Self {
         let keyboard = Robot::get_keyboard();
         keyboard.enable(SIMULATION_TIME_STEP);
-
-        Self { keyboard }
+        let receiver = Robot::get_receiver("ChestButton Channel");
+        receiver.enable(SIMULATION_TIME_STEP);
+        Self { keyboard, receiver }
     }
 }
 
 impl KeyboardDevice {
     pub fn get_touch_sensors(&self) -> TouchSensors {
         let key = self.keyboard.get_key();
+        let received_message = match self.receiver.get_next_packet() {
+            Ok(message) => message,
+            Err(error) => {
+                println!("error in received message: {error:?}");
+                None
+            }
+        };
+
         let control_shift_c_pressed = if let Some(key) = key {
             key == Keyboard::CONTROL | Keyboard::SHIFT | 'C' as u32
         } else {
-            false
+            received_message.is_some()
         };
```

Comment:
Doesn't this logic drop messages when any key is pressed?

Reply:
Technically yes, but irrelevant because if the keys are manually pressed then we can ignore presses caused by received messages

Reply:
You say "the keys", but we don't go into the else case even if the user is holding down some completely different key.